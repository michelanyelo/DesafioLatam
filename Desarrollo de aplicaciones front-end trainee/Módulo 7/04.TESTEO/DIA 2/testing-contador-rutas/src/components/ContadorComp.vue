<script setup>
import { ref } from "vue"

const contador = ref(0)

const aumentar = () => {
    contador.value++
}

const disminuir = () => {
    contador.value--
}
</script>

<template>
    <div>
        <h2>Contador</h2>
        <p id="contador">{{ contador }}</p>

        <button @click="disminuir" id="disminuir">-1</button>
        <button @click="aumentar" id="aumentar">+1</button>
    </div>
</template>

<style scoped></style>