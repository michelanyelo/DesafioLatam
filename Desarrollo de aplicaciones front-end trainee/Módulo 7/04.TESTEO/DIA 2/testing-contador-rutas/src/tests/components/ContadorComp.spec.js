import { describe, it, expect, test } from "vitest";
import { mount } from "@vue/test-utils";

import ContadorComp from "@/components/ContadorComp.vue";

describe("CounterComponent", () => {
    const wrapper = mount(ContadorComp)

    it("El html se mantiene en las pruebas", () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test("Tiene un boton para aumentar contador con texto '+1'", () => {
        const btnAumentar = wrapper.find("#aumentar")

        expect(btnAumentar.text()).toBe("+1")
    })

    test("Tiene un boton para disminuir contador con texto '-1'", () => {
        const btnDisminuir = wrapper.find("#disminuir")

        expect(btnDisminuir.text()).toBe("-1")
    })

    it("Aumenta el contador en uno al hacer click en '+1'", async () => {
        const contadorAntesClick = Number(wrapper.find("#contador").text())

        const btnAumentar = wrapper.find("#aumentar")

        await btnAumentar.trigger("click")

        const contadorDespuesClick = Number(wrapper.find("#contador").text())

        expect(contadorAntesClick).toBeLessThan(contadorDespuesClick)
    })

    it("Disminuye el contador en uno al hacer click en '-1'", async () => {
        const contadorAntesClick = Number(wrapper.find("#contador").text())

        const btnDisminuir = wrapper.find("#disminuir")

        await btnDisminuir.trigger("click")

        const contadorDespuesClick = Number(wrapper.find("#contador").text())

        expect(contadorDespuesClick).toBeLessThan(contadorAntesClick)
    })
})