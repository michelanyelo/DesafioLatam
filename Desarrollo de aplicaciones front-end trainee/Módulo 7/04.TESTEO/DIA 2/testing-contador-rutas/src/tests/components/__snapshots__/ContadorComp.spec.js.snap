// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CounterComponent > El html se mantiene en las pruebas 1`] = `
"<div>
  <h2>Contador</h2>
  <p id="contador">0</p><button id="disminuir">-1</button><button id="aumentar">+1</button>
</div>"
`;
